File: sarc/sarc.py
#! /usr/bin/env python

import sys

text: str = " ".join([arg.strip() for arg in sys.argv[1:]])

if not text:
    print("USAGE: sarc The text you want to be sarcastic")
    sys.exit(1)

should_upper_case = True
output = ""

for character in text:
    if character.isalpha():
        should_upper_case = not should_upper_case

    output = (
        output + character.upper() if should_upper_case else output + character.lower()
    )

print(output)

File: sarc/sarc.js
#! /usr/bin/env node

const [, , ...args] = process.argv;
const text = args.map((arg) => arg.trim()).join(" ");

if (!text) {
  console.log("USAGE: sarc The text you want to be sarcastic");
  process.exit(1);
}

let shouldUpperCase = true;
let output = "";

text.split("").forEach((character) => {
  const isAlpha = character.toLowerCase() != character.toUpperCase();
  if (isAlpha) {
    shouldUpperCase = !shouldUpperCase;
  }

  output += shouldUpperCase ? character.toUpperCase() : character.toLowerCase();
});

console.log(output);
